### PR TITLE
Setup CODEOWNERS

### PR DESCRIPTION
**Pull request by chmeliik**
CODEOWNERS will allow us to make the ownership of individual tasks
actually mean something.

We will have to give every owner write permissions, but we will be able
to set up branch protection rules to block PRs from getting merged if
they don't have approvals from the owners.

This will allow the CODEOWNERS of tasks and stepactions to fully
self-service their PRs without giving them the ability to mess with
other parts of the repository.

### CODE REVIEW

**Comment by chmeliik** [comment]
Changed the CODEOWNERS based on https://github.com/konflux-ci/build-definitions/pull/1658

Addressed the OWNERS_ALIASES vs. GH team mismatches
* `build-maintainers` matches the previous `build-team` now, minus the prefetch-only maintainers
* `ec` matches `ec-team` now
* `integration-service-maintainers` is `integration-team` + `chipspeak, jencull` (and those are integration team members, so that seems fine)

**Comment by chmeliik** [comment]
Need to add one more thing - Renovate groupings based on CODEOWNERS. Otherwise Renovate PRs will need approvals from many different sets of owners

**Comment by chmeliik** [comment]
CC @acornett21 @amisstea @arewm @avi-biton @bcrochet @brianwcook @brunoapimentel @cgwalters @eskultety @gbenhaim @gurnben @hmariset @jbpratt @kdudka @komish @ligangty @oamsalem @omeramsc @ralphbean @scoheb @skattoju @taylormadore @yftacherzog @yma96

This PR adds you as a CODEOWNER (because you were already in an OWNERS file), which means you'll now have the power to self-service changes to your tasks. On the other hand, it means your approval will be required in order for those changes to be merged. Let me know if you have any concerns about that.

**Comment by chmeliik** [comment]
Btw, the groupings based on filenames seem to work well. See for example https://github.com/chmeliik/build-definitions/pull/45, https://github.com/chmeliik/build-definitions/pull/46, which both update the oras image, but in two separate PRs according to the matchFileNames groupings

**Comment by chmeliik** [comment]
Added `/task/validate-fbc` and fixed the formatting (my auto-aligner script had a bug)

File: CODEOWNERS
Comment (yftacherzog) [inline CODEOWNERS:78]:
```suggestion
/task/provision-env-with-ephemeral-namespace    @amisstea @avi-biton @gbenhaim @omeramsc @yftacherzog
```

File: hack/update_renovate_json_based_on_codeowners.py
Comment (tnevrlka) [inline hack/update_renovate_json_based_on_codeowners.py:53]: nit: the `else` is unnecessary afaik

**Review by tnevrlka — APPROVED** [review]
🌮

File: CODEOWNERS
Comment (chmeliik) [reply CODEOWNERS:78]: :scream: good catch. @arewm we may have invited the wrong Omar into konflux-ci

File: hack/update_renovate_json_based_on_codeowners.py
Comment (chmeliik) [reply hack/update_renovate_json_based_on_codeowners.py]: Yup. Also reordered to avoid making the `return None` look like the happy path :smile:

File: CODEOWNERS
Comment (chmeliik) [reply CODEOWNERS]: Fixed. Went through the rest of the list, I think the rest are legit

File: CODEOWNERS
Comment (14rcole) [inline CODEOWNERS:60]: Both of these tasks should also have @kdudka as a maintainer

Comment (chmeliik) [reply CODEOWNERS]: Makes sense, added

**Comment by chmeliik** [comment]
There hasn't been pushback to this approach (or the pushback has been resolved). The CODEOWNERS file is now fully valid. Merging

**Pull request merged**
